File: AspNetCore/Reporting_ObjectDS_AspNetCore/PredefinedReports/ReportsFactory.cs
using DevExpress.XtraReports.UI;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Reporting_ObjectDS_AspNetCore
{
    public static class ReportsFactory
    {
        public static Dictionary<string, Func<XtraReport>> Reports = new Dictionary<string, Func<XtraReport>>()
        {
            ["EmployeeReport"] = () => new PredefinedReports.EmployeeReport()
        };
    }
}

File: MVC/VB/Reporting_ObjectDS_Mvc/PredefinedReports/ReportsFactory.vb
Imports DevExpress.XtraReports.UI
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Threading.Tasks

Namespace Reporting_ObjectDS_Mvc.PredefinedReports
	Public NotInheritable Class ReportsFactory

		Private Sub New()
		End Sub

		Shared Sub New()
			Reports.Add("EmployeeReport", Function() New EmployeeReport())
		End Sub
		Public Shared Reports As New Dictionary(Of String, Func(Of XtraReport))()
	End Class
End Namespace
